src/metabase/upload.clj

```diff
@@ -171,8 +171,16 @@
   (let [parsers (map #(upload-parsing/upload-type->parser % settings) col-upload-types)]
     (for [row rows]
       (for [[value parser] (u/map-all vector row parsers)]
-        (when-not (str/blank? value)
-          (parser value))))))
+        (do
+          (when-not parser
```

Comment:
water under the bridge right now since it's a backport, but in hindsight I think `if-not` > `do` + `when-not`

Reply:
Yeah, not sure why I didn't think of that.

I think `cond` would work even better, avoiding the nesting of conditionals.

```clojure
(cond 
  (not parser) (throw ...)
  (not (str/blank? value) (parser value))